test/e2e/app_suffix_test.go

```diff
@@ -0,0 +1,146 @@
+data:
+  key: value2
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config2
+data:
+  key: value
+`
+
+	logger.Section("deploy with 1 delete, 1 update, 1 create", func() {
+		kubectl.RunWithOpts([]string{"apply", "-f", "-"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(existingApp)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		NewPresentClusterResource("configmap", name+app.AppSuffix, env.Namespace, kubectl)
+		NewMissingClusterResource(t, "configmap", name, env.Namespace, kubectl)
+
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml2)})
+		NewPresentClusterResource("configmap", name+app.AppSuffix, env.Namespace, kubectl)
+		NewMissingClusterResource(t, "configmap", name, env.Namespace, kubectl)
+
+		cleanUp()
+	})
+
+	logger.Section("rename", func() {
+		newName := "test-app-suffix-app-exists-new"
+
+		kubectl.RunWithOpts([]string{"apply", "-f", "-"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(existingApp)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		kapp.Run([]string{"rename", "-a", name, "--new-name", newName})
+		NewPresentClusterResource("configmap", newName+app.AppSuffix, env.Namespace, kubectl)
+
+		kapp.Run([]string{"delete", "-a", newName})
+	})
+
+	logger.Section("delete", func() {
+		kubectl.RunWithOpts([]string{"apply", "-f", "-"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(existingApp)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		kapp.Run([]string{"delete", "-a", name})
+
+		NewMissingClusterResource(t, "configmap", name, env.Namespace, kubectl)
+	})
+
+	logger.Section("configmap with suffix exists and not marked as a kapp-app", func() {
+		fqName := name + app.AppSuffix
+
+		NewClusterResource(t, "configmap", fqName, env.Namespace, kubectl)
+		NewPresentClusterResource("configmap", fqName, env.Namespace, kubectl)
+
+		_, err := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+			RunOpts{IntoNs: true, AllowError: true, StdinReader: strings.NewReader(yaml1)})
+
+		require.Containsf(t, err.Error(), "kapp: Error:", "did not contain parseable app metadata")
+
+		RemoveClusterResource(t, "configmap", fqName, env.Namespace, kubectl)
+	})
+
+	logger.Section("does not migrate if USE_EXISTING_CONFIGMAP_NAME=True", func() {
+		os.Setenv("USE_EXISTING_CONFIGMAP_NAME", "True")
+
+		kubectl.RunWithOpts([]string{"apply", "-f", "-"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(existingApp)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
```

Comment:
I believe this is working right now by creating an app with new ConfigMap name and because there is no resource associated with the old ConfigMap, otherwise it would give ownership error for the resources present in the old app.